lib/material/extra/calendar/calendar.flow

```diff
@@ -924,21 +929,21 @@ dayEventsPopup(
 	isReadOnly = fgetValue(extractStruct(styles, CalendarReadOnly(const(false))).isReadOnlyB);
 	customPopupM = tryExtractStruct(styles, CalendarEventsPopup(nop4));
 	isReadOnlyEvent = extractStruct(styles, CalendarReadOnlyEvent(\__ -> const(false))).isReadOnly;
+	getHourOnClick = containsStruct(styles, CalendarGetHourOnEmptyCell());
 
 	close = make(false);
 	dayEvents = make(dayEventsStatic);
 
-	curTime = stamp2time(timestamp());
-	dayNumber = curDate.day;
-	monthNumber = curDate.month;
-
 	roundTime = \time -> {
-		roundInterval = 30;
-		roundedMinutes = (time.min / roundInterval + 1) * roundInterval;
-		flooredTime = Time(time with min = 0, sec = 0);
-		stamp2time(time2stamp(flooredTime) + min2ms(roundedMinutes));
+		if (getHourOnClick && (i2b(time.min) || i2b(time.sec))) {
```

Comment:
Shouldn't it be just ```if (!getHourOnClick) {```?

Reply:
hoursGridStyle = extractStruct(styles, HoursGridForWeekView(-1.0, 0, 24));
showHoursGrid = hoursGridStyle.cellSize > 0.0 && rangeType == 2;
getHourOnClick = containsStruct(styles, CalendarGetHourOnEmptyCell()) && showHoursGrid;

if we want to use only getHourOnClick i have to do more extractStruct that are not really neded